### DISCUSSION TITLE
Naming new experiment queue files

### DISCUSSION THREAD

**Discussion by cmmercer** [NMGRL/pychron#2195] [Q&A] [answered]
Is there a way to make the naming restrictions on new load file names less, well, restrictive? Currently, we can't name a file with two consecutive capital letters, or start a file name with numbers, or switch back and forth between numbers and letters.

**Comment by jirhiker** [comment]


**Reply by jirhiker** [reply]
You mean the name of the experiment queue?  

Those must follow the following PascalCase regex  `^[A-Z](([a-z0-9]+[A-Z]?)*)$`. 

If you'd like to make it less restrictive you can test out your own regexes at https://regex101.com/.  Once you are happy with the new regex I will incorporate in pychron

**Comment by cmmercer** [comment]
Yes, sorry, the names of experiment queues. We're finding the PascalCase a bit restrictive (we used to name experiments with leading numeric digits, and often have the name of our irrad. packages which have consecutive capital characters). 

I'll prototype another regex and get back to you!

**Comment by cmmercer** [comment]
What do you think of `^([0-9]+)?([A-Za-z0-9]+)$`? This allows leading numeric digits (optional), then expects alpha-numeric for the rest of the string (no enforcement of PascalCase).

**Comment by jirhiker** [comment, marked as answer]
How about `^[\w\d\-\_]+$`

12experiment  `good`
Experiment12  `good`
Experiment 12 `bad`
Experiemnt-12 `good`
Experiment_12 `good`
experiment12 `good`

**Reply by cmmercer** [reply]
That sounds good to me! Thanks!
